{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Import Dependencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "import json\n",
    "import datetime"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ---SUMMARY--- (From metadata.readme)\n",
    "#### total metadata rows: 44220\n",
    "#### custom_license: 16959 full text (new: 15533)\n",
    "#### noncomm_use_subset: 2353 full text (new: 385)\n",
    "#### comm_use_subset: 9118 full text (new: 128)\n",
    "#### biorxiv_medrxiv: 885 full text (new: 110)\n",
    "This adds up to 29,315. (This matches the files that can be found in the zipped files.) There are 44,226 rows in the metadata.csv, but 6 of these appear to be mis-applied and have author names. 15,759 rows in the metadata are missing the sha, or pdf hash. The \"has_full_text\" column has 28459 rows listed as \"True.\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\biorxiv_medrxiv\n",
      "paper_id\n",
      "metadata\n",
      "abstract\n",
      "body_text\n",
      "bib_entries\n",
      "ref_entries\n",
      "back_matter\n"
     ]
    }
   ],
   "source": [
    "## Checking out major keys/columns of an entry in the biorxiv folder\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\biorxiv_medrxiv\"\n",
    "\n",
    "## Opening JSON file\n",
    "read_file = open('00d16927588fb04d4be0e6b269fc02f0d3c2aa7b.json',)\n",
    "\n",
    "# returns JSON object as a dictionary \n",
    "data = json.load(read_file)\n",
    "\n",
    "# Iterating through the json list \n",
    "for i in data.keys(): \n",
    "    print(i) \n",
    "\n",
    "## Closiing file\n",
    "read_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\comm_use_subset\n",
      "The main keys for each json files are:\n",
      "paper_id\n",
      "metadata\n",
      "abstract\n",
      "body_text\n",
      "bib_entries\n",
      "ref_entries\n",
      "back_matter\n",
      "The key value pairs are:\n",
      "('paper_id', '000b7d1517ceebb34e1e3e817695b6de03e2fa78')\n",
      "('metadata', {'title': 'Supplementary Information An eco-epidemiological study of Morbilli-related paramyxovirus infection in Madagascar bats reveals host-switching as the dominant macro-evolutionary mechanism', 'authors': [{'first': 'Julien', 'middle': [], 'last': 'Mélade', 'suffix': '', 'affiliation': {'laboratory': \"Centre de Recherche et de Veille sur les Maladies Emergentes dans l'Océan Indien (CRVOI), Plateforme de Recherche CYROI\", 'institution': '', 'location': {'addrLine': '2 rue Maxime Rivière, 97490 Sainte Clotilde, La Réunion', 'country': 'France'}}, 'email': ''}, {'first': 'Nicolas', 'middle': ['Wieseke'], 'last': '4#', 'suffix': '', 'affiliation': {'laboratory': '', 'institution': 'University of Leipzig', 'location': {'addrLine': 'Augustusplatz 10', 'postCode': 'D-04109', 'settlement': 'Leipzig', 'country': 'Germany'}}, 'email': ''}, {'first': 'Beza', 'middle': [], 'last': 'Ramazindrazana', 'suffix': '', 'affiliation': {'laboratory': \"Centre de Recherche et de Veille sur les Maladies Emergentes dans l'Océan Indien (CRVOI), Plateforme de Recherche CYROI\", 'institution': '', 'location': {'addrLine': '2 rue Maxime Rivière, 97490 Sainte Clotilde, La Réunion', 'country': 'France'}}, 'email': ''}, {'first': 'Olivier', 'middle': [], 'last': 'Flores', 'suffix': '', 'affiliation': {'laboratory': 'UMR C53 CIRAD, Peuplements Végétaux et Bioagresseurs en Milieu Tropical', 'institution': '', 'location': {'addrLine': \"7 chemin de l'IRAT\", 'postCode': '97410', 'settlement': 'St Pierre', 'country': 'France'}}, 'email': ''}, {'first': '$', 'middle': [], 'last': '', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Erwan', 'middle': [], 'last': 'Lagadec', 'suffix': '', 'affiliation': {'laboratory': \"Centre de Recherche et de Veille sur les Maladies Emergentes dans l'Océan Indien (CRVOI), Plateforme de Recherche CYROI\", 'institution': '', 'location': {'addrLine': '2 rue Maxime Rivière, 97490 Sainte Clotilde, La Réunion', 'country': 'France'}}, 'email': ''}, {'first': 'Yann', 'middle': [], 'last': 'Gomard', 'suffix': '', 'affiliation': {'laboratory': \"Centre de Recherche et de Veille sur les Maladies Emergentes dans l'Océan Indien (CRVOI), Plateforme de Recherche CYROI\", 'institution': '', 'location': {'addrLine': '2 rue Maxime Rivière, 97490 Sainte Clotilde, La Réunion', 'country': 'France'}}, 'email': ''}, {'first': 'Steven', 'middle': ['M'], 'last': 'Goodman', 'suffix': '', 'affiliation': {}, 'email': ''}, {'first': 'Koussay', 'middle': [], 'last': 'Dellagi', 'suffix': '', 'affiliation': {'laboratory': \"Centre de Recherche et de Veille sur les Maladies Emergentes dans l'Océan Indien (CRVOI), Plateforme de Recherche CYROI\", 'institution': '', 'location': {'addrLine': '2 rue Maxime Rivière, 97490 Sainte Clotilde, La Réunion', 'country': 'France'}}, 'email': ''}, {'first': 'Hervé', 'middle': [], 'last': 'Pascalis', 'suffix': '', 'affiliation': {'laboratory': \"Centre de Recherche et de Veille sur les Maladies Emergentes dans l'Océan Indien (CRVOI), Plateforme de Recherche CYROI\", 'institution': '', 'location': {'addrLine': '2 rue Maxime Rivière, 97490 Sainte Clotilde, La Réunion', 'country': 'France'}}, 'email': 'herve.pascalis@ird.fr.'}]})\n",
      "('abstract', [])\n",
      "('body_text', [{'text': '- Figure S1 : Phylogeny of all sequences belonging to the UMRV phylogroup. - Table S4 : Bats cytochrome b sequences data set. -Table S5 : Test of host-parasite co-evolution using global fit methods ParaFit. Figure S1 . Phylogeny of all sequences belonging to the UMRV phylogroup. A global phylogeny of 308 partial L-gene sequences calculated in 10,000,000 iterations in MrBayes with the GTR + G + I evolutionary model and a 10% burn-in rooted with an Aquaparamyxovirus sequence (GenBank number EF646380). All Malagasy bat paramyxoviruses sequences obtained within this study fell within group of Unclassified Morbillivirus-Related viruses. Genbank accession numbers used for each virus genera are indicated in Table S6 . ', 'cite_spans': [], 'ref_spans': [{'start': 2, 'end': 11, 'text': 'Figure S1', 'ref_id': None}, {'start': 77, 'end': 85, 'text': 'Table S4', 'ref_id': None}, {'start': 126, 'end': 135, 'text': '-Table S5', 'ref_id': None}, {'start': 207, 'end': 216, 'text': 'Figure S1', 'ref_id': None}, {'start': 710, 'end': 718, 'text': 'Table S6', 'ref_id': None}], 'section': ''}])\n",
      "('bib_entries', {'BIBREF32': {'ref_id': 'b32', 'title': 'NDV/HQ266603/Chicken/1992', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF43': {'ref_id': 'b43', 'title': 'MuV/FJ375177/Human/1987', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF48': {'ref_id': 'b48', 'title': 'HeV/HM044321/Horse/2007', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF72': {'ref_id': 'b72', 'title': 'NDV/FJ751918/Chicken/1979', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF87': {'ref_id': 'b87', 'title': 'APMV4/EU877976/Duck/2006', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF123': {'ref_id': 'b123', 'title': 'NiV/FJ513078/Homo_sapiens/2007', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF127': {'ref_id': 'b127', 'title': 'MuV/AF280799/Monkey/1996', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF151': {'ref_id': 'b151', 'title': 'CDV/KF914669/Dog/2013', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF153': {'ref_id': 'b153', 'title': 'APMV2/HM159993/Chicken/2006', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF158': {'ref_id': 'b158', 'title': 'Salem_virus/JQ697837/Horse/1992', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF163': {'ref_id': 'b163', 'title': 'APMV2/EU338414/Chicken/1956', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF170': {'ref_id': 'b170', 'title': 'APMV8/FJ215863/Goose/1976', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF218': {'ref_id': 'b218', 'title': 'HeV/HM044319/Horse/2007', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}, 'BIBREF223': {'ref_id': 'b223', 'title': 'MeV/GQ376026/Human/1999', 'authors': [], 'year': None, 'venue': '', 'volume': '', 'issn': '', 'pages': '', 'other_ids': {}}})\n",
      "('ref_entries', {'TABREF0': {'text': 'Table S4. Bats cytochrome b sequences data set. Sequence generated as part of this study is underlined. Miniopterus gleni; N° 10: Miniopterus sororculus; N° 11: Miniopterus mahafaliensis; N° 12: Miniopterus manavi; N° 13: Miniopterus griveaudi.', 'latex': None, 'type': 'table'}})\n",
      "('back_matter', [])\n"
     ]
    }
   ],
   "source": [
    "## Doing the same has above, but for the commercial use subset\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\comm_use_subset\"\n",
    "\n",
    "## Opening JSON file\n",
    "read_file = open('000b7d1517ceebb34e1e3e817695b6de03e2fa78.json',)\n",
    "\n",
    "# returns JSON object as a dictionary \n",
    "data = json.load(read_file)\n",
    "\n",
    "# Iterating through the json list \n",
    "print(\"The main keys for each json files are:\")\n",
    "for i in data.keys(): \n",
    "    print(i) \n",
    "\n",
    "\n",
    "print(\"The key value pairs are:\")\n",
    "for i in data.items():\n",
    "    print(i)\n",
    "\n",
    "## Closing file\n",
    "read_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\biorxiv_medrxiv\n"
     ]
    }
   ],
   "source": [
    "## We know the key names, so we can extract the values into objects\n",
    "## Looking at the file itself in notepad, we can also extract the title\n",
    "\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\biorxiv_medrxiv\"\n",
    "## Opening JSON file\n",
    "read_file = open('00d16927588fb04d4be0e6b269fc02f0d3c2aa7b.json',)\n",
    "\n",
    "# returns JSON object as a dictionary \n",
    "data = json.load(read_file)\n",
    "\n",
    "paper_id = data['paper_id']\n",
    "title = data['metadata']['title']\n",
    "\n",
    "abstract_str = ''\n",
    "for text in data['abstract']:\n",
    "    abstract_str = abstract_str + ' ' + text['text'] \n",
    "\n",
    "body_text_str = ''\n",
    "for text in data['body_text']:\n",
    "    body_text_str = body_text_str + ' ' + text['text']\n",
    "\n",
    "bib_entries = data['bib_entries']\n",
    "ref_entries = data['ref_entries']\n",
    "back_matter = data['back_matter']\n",
    "\n",
    "## Closing file\n",
    "read_file.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' Infectious bronchitis (IB), which is caused by infectious bronchitis virus (IBV), is one of the most important diseases of poultry, causing severe economic losses worldwide. 8 Clinical signs of disease are diverse and include respiratory distress, severe ocular discharge, poor body weight gain, decreased egg production, flushing (renal disease), and occasionally mortality in chickens. 7 IB is often complicated by secondary bacterial (e.g., E. coli, Mycoplasma sp.) and viral infections (e.g., avian influenza virus, Newcastle disease virus, and infectious laryngotracheitis virus). 43 Lack of cross protection among IBV serotypes is a challenge to controlling IB 15 ; therefore, control of IB relies heavily on serotype-specific live attenuated vaccines. 8 Collectively, the presence of multiple IBVs in a single sample, emergence of variant IBVs, and the high genetic diversity of IBV can complicate the diagnosis of IB and illustrate the need for enhanced diagnostics. 15  Clinical swab samples (n = 15) were obtained from samples submitted to the Poultry Diagnostic & Research Center, University of Georgia. Detailed information about the samples is provided in Table 1 . Total RNA was extracted from each of the swab samples using QIAamp Viral RNA Mini kit (Qiagen, Hilden, Germany) as per manufacturer\\'s instructions, aliquoted, and stored at −80 ºC until further use. These samples were previously tested for IBV and GA08 serotype of IBV. Briefly, a pan IBV RT-qPCR assay 4 was used to detect IBV in general and then IBV strain specific RT-qPCR assay 36 was used to detect GA08 in the samples using the AgPath ID, One step RT-PCR kit (Thermo Fisher Scientific, Waltham, MA, USA) on the Applied Biosystems 7500 FAST following the previously described protocols. For MinION cDNA synthesis, RNA was extracted from 500 µl of each clinical sample using Trizol-LS (Thermo Fisher Scientific, Waltham, MA, USA), per manufacturer\\'s directions. A reaction mixture of 8 µl of total RNA, 1 µl of random primers and 1 µl of dNTPs was incubated at 65 °C for 5 min, chilled on ice for at least 1 min followed by addition of 10 µl of cDNA synthesis mix including SuperScript III (Thermo Fisher Scientific, Waltham, MA, USA) according to the manufacture\\'s instruction. The reaction was incubated at 25 °C for 10 min, then at 50 °C for 50 min for cDNA synthesis. The reaction was terminated at 85 °C for 5 min, and then chilled on ice. To remove residual RNA, the cDNA solution was incubated with RNase H at 37 °C for 20 min. All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. sequencing protocol was initiated using the MinKNOW software v.5.12. Sequencing was allowed to continue for 2 hours until 42,940 reads were obtained for Run1 and 156,000 reads were obtained for Run 2 in 6 hours. First, a lineage-typing database, containing 32 IBV S1 gene sequences (one sequence from each of 32 lineages in the 6 genotypes 44 ) and the chicken genome (GCF_000002315.4_Gallus_gallus-5.0), was constructed (referred to as \"IBV-lineage-typing database\" in this study). A second database was constructed with all the Avian Coronaviruses S1 sequences (n = 7328) available in NCBI (as of: 09-08-2017) and the chicken genome (referred to as \"All-IBV database\" in this study). Prior to database construction, all sequences were dustmasked (NCBI C++ ToolKit), and each isolate was assigned a unique taxonomy ID as a species hierarchically under the genus Gammacoronavirus to allow easy sorting. The local BLAST databases were compiled using default settings. The IBV-lineage-typing database was All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint used to assign IBV lineages to each read, as appropriate. The All-IBV database was used to assign IBV isolates taxonomic ID to each of IBV reads from within IBV lineage read clusters. The IBV-lineage-typing database, which contains only one sequence per lineage, is required as centrifuge divides the score for any given read by the number of hits that have an equal score. Since some lineages are overrepresented in GenBank (e.g., Mass strains), this results in the scores of those reads (e.g., Mass reads) being divided by a large number, effectively reducing the score for any single alignment. However, because there is only one read per lineage in the IBVlineage-typing database, it has insufficient diversity to determine if there is more than one strain of the same lineage present in a sample. Thus, the All-IBV database provides the diversity required for that analysis. A schematic diagram of the workflow of MinION data analysis is presented in Fig. 1 . For Nanopore sequencing data, pre-processing steps were performed to prepare data for downstream analysis. Briefly, Nanopore reads (FAST5) were basecalled using Albacore v 2.02 (https://github.com/Albacore/albacore) with following parameters (read_fast5_basecaller.py -i /Input_reads.fast5 --recursive -t 4 -s /Output_files --flowcell FLO-MIN107 --kit SQK-LSK108o fastq). Porechop (https://github.com/rrwick/Porechop) was used for adapter trimming, (default setting), barcode-based demultiplexing (default settings), and to trim an additional 21 nucleotides, representing the S1 primer sequences (porechop -i Input_file.fastq --extra_end_trim 21 -b ./output_demultiplexed files.fastq). After barcode and adapter removal, reads were analyzed with a script-based, 2-step data analysis protocol, which includes Centrifuge-kreport as taxonomic read classifier 24 using the sequences in the above mentioned BLAST databases. Briefly, basecalled reads (FASTQ) from individual barcoded samples were used as an input. All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint First, the basecalled reads were aligned to the IBV-lineage-typing database using BLASTn and reads were clustered based on the read sequence alignment to the respective prototype sequence of IBV lineage. These read clusters were used to interpret presence of IBV genotypes and lineages in the samples. For isolate level identification of IBV, the lineage-based read clusters were individually aligned to the All-IBV database, which produced sub-clusters of reads. Each of these read sub-clusters potentially represents a different isolate and was further used for interpretation. Knowing that MinION sequencing has a high sequencing error rate in individual reads, further steps were added in the data analysis algorithm to obtain a more accurate consensus sequence. Therefore, these read sub-clusters were mapped, using Geneious mapper in Geneious software v11.1.3. (Biomatters, Ltd, Auckland, New Zealand), to the IBV-lineagetyping database (a reference FASTA file) to obtain consensus sequences from each sub-cluster. A minimum threshold for number of reads per consensus sequence was not set as in two samples (sample #4 in Run1 and sample #12 in Run 2), only 1 and 5 IBV reads were obtained, respectively; therefore, consensus sequences were built from all the available IBV reads per subcluster. The consensus sequences were compared to GenBank using BLASTn. To select the \"top hit\" from BLASTn output, sequence search results were ordered by \"sequence identity\" and then sequence alignments were evaluated for \"minimum mismatches\" and \"coverage\" of query or subject sequence. All isolates with the highest query or subject coverage and the fewest mismatches were used as \"top hit\" in the final results. For Sanger sequencing amplicons synthesis, cDNA from MinION library preparation was amplified using the following primer sets (Table 2) . Firstly, a primer set based on Connecticut sequence NCBI FJ904716.1 was used to amplify Genotype I-lineage 1 viruses (e.g., Connecticut All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint and Massachusetts serotypes). Secondly, a primer set based on GA98 sequence NCBI AF274439.1 was used to amplify Genotype IV-lineage 1 viruses (e.g., Georgia 1998 and Delaware 072 serotypes). Primer sets were designed using NCBI Primer-BLAST. 51 The PCR reaction mixture is composed of 10 µl of cDNA, 0.5 µl of 10 µM forward primer, 0.5 µl of 10 µM reverse primer, 2.5 µl of 10× buffer, 0.5 µl of enzyme mix, and 0.5 µl of 10mM dNTPs, then made the final volume to 25 µl with nuclease-free water. The following thermocycling conditions were used for amplicon synthesis: denaturation at 95 °C for 1 min, followed by  For Sanger sequencing data, the chromatogram from each of the samples was manually checked and primer sequences were trimmed in MEGA 6.0. 42 Forward and reverse sequences from multiple clones were aligned using MEGA 6.0 software and consensus sequences were All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint compared to GenBank using Basic Local Alignment Tool (BLAST) (https://blast.ncbi.nlm.nih.gov/) (as of: 12-03-2018) and the top hit from BLASTn output was selected, as described above for MinION sequencing data, to identify the IBV isolate in samples. A pairwise nucleotide identity comparison on the partial S1 sequences of IBVs from each of the samples sequenced with Sanger sequencing and MinION sequencing was performed. Briefly, the final AmpSeq consensus sequences from identical IBV types in each of the sample were aligned using ClustalW in MEGA6 42 and this alignment was used for pairwise nucleotide identity using MEGA6. 42 All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint Five samples from Run 1 were tested with a pan-IBV RT-qPCR assay 4 , and four of those were positive for IBV. Samples with pathogenic IBV strains (Run 2, n = 5) were tested with the pan-IBV RT-qPCR and all five samples were IBV positive (Table 1) . Next, a GA08 serotype specific primer set (in house validated set of primers used for diagnostics) was used on all samples from Run 1 (including five samples that were previously tested by the pan-IBV PCR) and four of 10 samples were positive. Sample #3 and #4, which were positive with the pan-IBV RT-qPCR assay, were negative with GA08 specific RT-qPCR assay indicating that these samples contained IBV serotypes other than GA08 and required further testing. Sample #5 tested negative with both assays (Table 1) . A total of 10 clinical swab samples (Run 1) were processed for Sanger sequencing to confirm the presence of IBV lineages. Based on the MinION results, primers targeting GIV-L1 and GI-L1 (GA98 and Conn and primer sets, respectively) were created. The PCR products were cloned and multiple (6-24) clones from each sample were submitted for Sanger sequencing and results are summarized in Table 3 . Using GIV-L1 serotype primers showed that 9 out of 10 samples were positive with 1 sample (#5) negative for IBV (which is consistent with the pan-IBV RT-qPCR results for this sample). Consensus sequences from multiple clones obtained by using GIV-L1 primers showed the presence of GA98 (GA/A9dvaccinated) in 7 of 9 IBV-positive samples. DE072 was detected in samples #1 and #9 as the lone GIV-1 isolate but was also detected in three samples (#2, and #10) that also contained GA98 (Table 4 ). By using the GI-L1 primer set, which amplified Conn and Mass type IBVs, 3 of 9 IBV-positive samples were All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. (Table 4 ). Based on the two conventional RT-PCRs and the GA08 RT-qPCR, it was determined that only samples #8 and #9 contained a single IBV isolate, while the other 7 samples were positive for 2-4 types of IBV (Table 3) . PCR amplicons obtained directly from 10 clinical swab samples with vaccine IBV serotypes were barcoded, pooled together and sequenced on MinION device (Run 1). A total of 38,661 reads were successfully basecalled from the entire sequencing run (total reads = 42,940). After demultiplexing, reads per barcode ranged from 831 to 4114. A total of 14,845 reads were not assigned to any of the used barcodes and 128 reads were discarded due to middle adapters. The Nanopore reads were queried against IBV-lineage-typing database to determine if the samples contained IBV (Figure 1 ). This AmpSeq protocol detected IBV reads in 8 of 10 samples. The number of IBV reads per sample (IBV positive sample) ranged from 56-944. The MinIONnegative samples included sample #5, which was consistently negative with RT-qPCR and RT-PCR assays (and thus interpreted as IBV negative), and sample #1, which had the highest Cq values in the pan-IBV RT-qPCR (Table 1) . Sequencing data was further analyzed to determine if multiple IBV genotypes or lineages could be detected. Lineage-1 and lineage-27 from Genotype I (GI-L1, GI-L27), and lineage-1 from Genotype IV (GIV-L1) were the detected lineages in the samples. Results from individual RT-qPCR assay for GA08 (GI-L27), and RT-PCR assays for GIV-L1and GI-L1 confirmed the presence of multiple IBV genotypes and lineages in 6 of the 9 IBV-positive samples (#1, #2, #3, #4, #6, #7) and AmpSeq results confirmed multiple genotypes and lineages in 4 (#2, #4, #6, #7) of those 6 samples (AmpSeq failed to detect any IBV in #1, and failed to detect GIV-L1 in #3). All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint The presence of a single lineage was confirmed by PCR-based assays and AmpSeq in three samples (#8, # 9 and #10) ( Table 3) . It was observed that the single read cluster composed of less than 5 reads (Run1, sample #4, GA98) yielded a poor-quality consensus sequence, consistent with the known individual read error rate of MinION sequencing. High-quality consensus sequences were obtained from the other read clusters (>5 reads per cluster). Each of the obtained consensus sequences were compared to GenBank databases using BLAST, which revealed >99% sequence identity to respective IBV isolates Table 4 ). As described in the Sanger Sequencing section, two samples (#8 and #9) contained only one IBV isolate per the non-MinION assays, and the AmpSeq results were consistent with these findings. Non-MinION assays showed that seven samples contained two or more isolates per sample. Six samples (#1, #3, #4, #6, #7, and #10) contained two isolates per sample (Table 4 ). Of those six samples, AmpSeq detected both isolates in four samples (#4, #6, #7, and #10), one isolate in one sample (#3), and no IBV in one sample (#1; as mentioned above, sample #1 was the sample with the highest Cq value). Of note, sample #10 contained two isolates from the same lineage and AmpSeq was able to identify both isolates within this sample. Finally, the seventh multi-isolate sample (#2) contained four isolates per the non-MinION results, and the AmpSeq detected two of those four isolates. These data show that a co-infection of multiple IBV lineages existed in the above-mentioned samples but a single RT-qPCR, or Sanger sequencing of a single clone may not have detected these co-infecting IBV lineages and required multiple assays to detect all the IBV types. However, the AmpSeq protocol accurately detected multiple IBV lineages in 4 of 7 samples, with partial detection in two of the remaining three samples (Table 4 ). In samples which had the same IBV type, Blast search of consensus All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint sequences obtained from AmpSeq and Sanger sequencing identified the same (or highly related isolate for sample #6 [GA98] and sample #9) in the NCBI nt database as per parameters described in the methods section. To evaluate the utility of this protocol on clinical swab samples with pathogenic IBV variants, IBV-S1 gene was amplified directly from five clinical tracheal scrapings and the PCR amplicons were used to create MinION libraries. A total of 146,540 reads were successfully basecalled from the entire sequencing run (total reads = 156,000). After demultiplexing, reads per barcode ranged from 4,285 to 41,131. A total of 24,297 reads were not assigned to any of the used barcodes and 8,109 reads were discarded due to middle adapters in the basecalled reads. Real-time analysis of MinION data, which was obtained within 10 minutes of the sequencing, was sufficient for detection of IBV. However, the sequencing data obtained from entire sequencing run was processed with the same protocol as described above. This AmpSeq protocol detected IBV reads in all of the 5 tested samples using all basecalled reads. The number of IBV reads per sample ranged from 5-4,956. Additionally, the sequencing data analysis showed that the IBV reads belonged to GI-17 lineage and typed the isolate as DMV1639. After MinION sequencing results, these samples were later tested to confirm the presence of IBV variant by MDL_DMV1639 IBV variant specific RT-qPCR assay. All 5 samples were positive for IBV MDL_DMV1639 variant of IBV (Table 5) . In all of the samples where a matching IBV genotype/lineage was identified by both AmpSeq and Sanger sequencing, the AmpSeq data showed high concordance with Sanger sequencing data, In one sample (#4), only a single MinION read of GA98 was detected by All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint sequencing, consistent with the reported single-read accuracy of MinION sequencing. All other samples showed 99-100% pairwise identity across the partial S1 fragment generated by Sanger Sequencing. (Table 4 ). All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint This study describes a sequencing approach for rapid and accurate identification of IBV genotypes/lineages utilizing the MinION Nanopore sequencer. Additionally, a sequence analysis workflow is described here. The workflow analysis is designed to deal with the multiple, highly similar sequences contained within GenBank that are due to numerous sequencing experiments of IBV (especially vaccine viruses). This protocol demonstrates the feasibility of real-time-based sequencing to provide isolate-level resolution of IBV directly from clinical swab samples, including samples that contain multiple isolates. The proper diagnosis of IBV as the cause of clinical respiratory disease is contingent on virus typing and differentiating live vaccine viruses from field strains. 11 Previously, it was reported that IBV genotypes correlate well with the serotypes of IBV 45 ; therefore, accurate genotypic identification of IBV will be useful to diagnose vaccine and variant viruses in clinical samples. The rapid pan-IBV and serotype specific RT-qPCR assays 4,23,36 have been used for serotype identification; however, positive results from RT-qPCR is insufficient to determine the IBV genotype and obtain isolate-level resolution of IBV; thus, sequence analysis of IBV-S1 gene is required. Previously, use of partial S1 gene sequences (450 bp) to type IBV has been described. 29 However, increasing the length of sequenced S1 gene (~1620 bp) results in more data to be used for genotyping as more of the hypervariable region is covered. 17, 27, 45 For example, the AmpSeq protocol was able to differentiate two genetically very similar (99.5% at S1 gene of IBV genome) but serotypically different GIV-L1 isolates, DE072 and GA98, within a single sample. Thus, the detection of highly diverse IBV genotypes, via the S1 subunit, with a single and rapid sequencing protocol is desirable and the ability to detect multiple isolates in a single sample would improve IBV detection in clinical samples. All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint Currently, pan-IBV RT-qPCR is used to rapidly detect IBV from clinical samples for screening purposes, 4 but this requires additional serotype-specific RT-qPCRs to genotype positive samples, including samples containing more than one genotype. Additionally, if unidentified IBVs are in the sample, then current detection and characterization may also require egg culture followed by various PCRs. 8 The AmpSeq method described in this report detected IBV in 13 of 14 IBV-positive and detected all the mixed isolates in four of seven samples containing two or more isolates. While mixed isolates were not 100% identified (0/2 in #1, 2/4 in #2, and ½ in #3) in three samples with the AmpSeq method, detection of those mixed isolates required several non-AmpSeq assays. Although, there is room for improving this new AmpSeq method, it represents a promising, single-step assay that can be used without egg culture. Thus, it is another tool for the detection of IBV, especially in cases when multiple isolates may be present or when genotyping is especially important. One area that is problematic for a test such as AmpSeq is to detect all the isolates that may be present. While AmpSeq did detect all isolates in 4 samples, in 3 cases not all of the isolates were detected. A potential explanation to partial detection of multiple IBV genotypes could be the relative abundance of IBV genotypes in these clinical samples. It could also be that amplification of these IBV genotypes by serotype-specific RT-qPCR assays is more efficient due to small targeted fragment size (e.g., 120 bp for PCR and ~1600 bp for AmpSeq) and better primer alignment to the target (e.g., degenerate bases are used in the S1 primers used for AmpSeq). One complicating factor of the current AmpSeq protocol is that the IBV target sequence 27 was not originally designed for high specificity, especially from clinical samples. As such, a high proportion of total reads were non-IBV reads (e.g., often mapping to the chicken genome, data not shown), consistent with the extra bands visible in the original report for these All rights reserved. No reuse allowed without permission. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint target sequences. 28 Additionally, it is possible that certain genotypes are better complemented to the S1 primers than others and may outcompete those isolates for amplification in the AmpSeq protocol. Lastly, increasing the total number of reads collected by AmpSeq may improve the ability to detect all genotypes in sample by AmpSeq. It is possible to allow the sequencing to continue longer to obtain more reads per sample. Overall, AmpSeq is a feasible test for IBV characterization, and work is ongoing to improve this new type of assay. For example, cost and time efficiency of a sequencing protocol can be improved through multiplexing of more samples in a single sequencing run. 3, 47 In this study, samples (Run 1 = 10, Run 2 = 5) were simultaneously multiplexed (i.e., pooled and then sequenced in one run) while maintaining IBV genotyping from data collected. Since the MinION flow cells were not exhausted, and can be washed and re-used, the AmpSeq method also has the potential for sequential multiplexing. This would decrease the need to hold samples for weeks while waiting for the cost-optimal number of samples for simultaneous multiplexing. Alternatively, if detection of all isolates within a given sample is preferred, the sequencing can be run longer to increase sensitivity. The single protocol nature of AmpSeq, the ability to obtain S1 gene sequence results, real-time data analysis, and flexibility of testing design makes MinION-based AmpSeq a viable sequencing protocol for typing of IBV isolates. The advent of real-time, in-house, third generation sequencing represents a transformative opportunity for diagnostic laboratories, by offering the ability to more fully characterize PCR reactions beyond confirming amplicon size (e.g., routine electrophoresis), Sanger sequencing RT-PCR products, or by confirming a partial sequence through probe hybridization (e.g., probe-based qPCR). However, the interpretation of such large data sets represents a challenge to veterinary diagnosticians. Read-based classification software such as genotypes from single sample. Thus, the MinION-based AmpSeq coupled with the data analysis workflow for identification, differentiation, and accurate prediction of IBV genotypes from clinical swab samples can be used as an adjunct to other established rapid diagnostic assays 9,12 until extensive testing of this protocol is done to improve and validate AmpSeq for IBV diagnostics. Furthermore, AmpSeq-based assays can be, and are being applied to other viral pathogens, 3,37 demonstrating the power and utility of this method in the age of molecular diagnostics. The copyright holder for this preprint (which was not peer-reviewed) is the author/funder. . https://doi.org/10.1101/634600 doi: bioRxiv preprint '"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##Example of a body texxt extracted from a file\n",
    "body_text_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\n"
     ]
    }
   ],
   "source": [
    "# -----------------------------------------------------------\n",
    "# Importing all the file names into lists\n",
    "# -----------------------------------------------------------\n",
    "\n",
    "\n",
    "\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\"\n",
    "\n",
    "import os\n",
    "\n",
    "bio_med_files = []\n",
    "comm_files = []\n",
    "custom_files = []\n",
    "noncomm_files = []\n",
    "\n",
    "for f in os.listdir(\"biorxiv_medrxiv\"):\n",
    "    bio_med_files.append(f)\n",
    "for f in os.listdir(\"comm_use_subset\"):\n",
    "    comm_files.append(f)\n",
    "for f in os.listdir(\"custom_license\"):\n",
    "    custom_files.append(f)\n",
    "for f in os.listdir(\"noncomm_use_subset\"):\n",
    "    noncomm_files.append(f)\n",
    "    \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Length of bio_med_files: 885 \n",
      "Length of comm_files: 9118 \n",
      "Length of custom_files: 16959 \n",
      "Length of noncomm_files: 2353 \n"
     ]
    }
   ],
   "source": [
    "## Lengths of files match up to readme and directory count\n",
    "print(\"Length of {}: {} \".format(\"bio_med_files\", len(bio_med_files)))\n",
    "print(\"Length of {}: {} \".format(\"comm_files\", len(comm_files)))\n",
    "print(\"Length of {}: {} \".format(\"custom_files\", len(custom_files)))\n",
    "print(\"Length of {}: {} \".format(\"noncomm_files\", len(noncomm_files)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2020-03-26 18:41:09.795413\n",
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\biorxiv_medrxiv\n",
      "2020-03-26 18:41:10.989616\n",
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\comm_use_subset\n",
      "2020-03-26 18:41:28.644970\n",
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\custom_license\n",
      "2020-03-26 18:42:20.022363\n",
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\noncomm_use_subset\n",
      "2020-03-26 18:42:26.238136\n"
     ]
    }
   ],
   "source": [
    "# -----------------------------------------------------------\n",
    "# Creating lists for the different key value pairs in the json dictionary\n",
    "# -----------------------------------------------------------\n",
    "\n",
    "##defining function so that I don't repeat myself\n",
    "\n",
    "## We also only want the paper id, title, abstract, and actual text of the paper\n",
    "def json_reader(r):\n",
    "    read_file = open(r,)\n",
    "    data = json.load(read_file)  ## data is a dictionary\n",
    "    paper_id.append(data['paper_id'])\n",
    "    title.append(data['metadata']['title'])\n",
    "    \n",
    "    abstract_str = ''\n",
    "    for text in data['abstract']:\n",
    "        abstract_str = abstract_str + ' ' + text['text'] \n",
    "\n",
    "    body_text_str = ''\n",
    "    for text in data['body_text']:\n",
    "        body_text_str = body_text_str + ' ' + text['text']\n",
    "\n",
    "    abstract.append(abstract_str)\n",
    "    body_text.append(body_text_str)\n",
    "\n",
    "    read_file.close()\n",
    "    return\n",
    "    \n",
    "paper_id = []\n",
    "title = []\n",
    "abstract = []\n",
    "body_text = []\n",
    "\n",
    "file_location = []\n",
    "\n",
    "\n",
    "print(datetime.datetime.now())\n",
    "\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\biorxiv_medrxiv\"\n",
    "for r in bio_med_files:\n",
    "    json_reader(r)\n",
    "    file_location.append(\"biorxiv_medrxiv\")\n",
    "\n",
    "print(datetime.datetime.now())\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\comm_use_subset\"\n",
    "for r in comm_files:\n",
    "    json_reader(r)\n",
    "    file_location.append('comm_use subset')\n",
    "\n",
    "print(datetime.datetime.now())\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\custom_license\"\n",
    "for r in custom_files:\n",
    "    json_reader(r)\n",
    "    file_location.append(\"custom_license\")\n",
    "\n",
    "print(datetime.datetime.now())\n",
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\\noncomm_use_subset\"\n",
    "for r in noncomm_files:\n",
    "    json_reader(r)\n",
    "    file_location.append(\"noncomm_use_subset\")\n",
    "\n",
    "print(datetime.datetime.now())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "29315"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## The length of the lists match the number of papers from the top of the program\n",
    "len(paper_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Creating data frame from the \"scrapped\" information\n",
    "df = pd.DataFrame({'paper_id': paper_id, 'title': title, 'abstract': abstract,'body': body_text, 'origin': file_location})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(29315, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>paper_id</th>\n",
       "      <th>title</th>\n",
       "      <th>abstract</th>\n",
       "      <th>body</th>\n",
       "      <th>origin</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0015023cc06b5362d332b3baf348d11567ca2fbb</td>\n",
       "      <td>The RNA pseudoknots in foot-and-mouth disease ...</td>\n",
       "      <td>word count: 194 22 Text word count: 5168 23 2...</td>\n",
       "      <td>VP3, and VP0 (which is further processed to V...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>004f0f8bb66cf446678dc13cf2701feec4f36d76</td>\n",
       "      <td>Healthcare-resource-adjusted vulnerabilities t...</td>\n",
       "      <td></td>\n",
       "      <td>The 2019-nCoV epidemic has spread across Chin...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>00d16927588fb04d4be0e6b269fc02f0d3c2aa7b</td>\n",
       "      <td>Real-time, MinION-based, amplicon sequencing f...</td>\n",
       "      <td>Infectious bronchitis (IB) causes significant...</td>\n",
       "      <td>Infectious bronchitis (IB), which is caused b...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0139ea4ca580af99b602c6435368e7fdbefacb03</td>\n",
       "      <td>A Combined Evidence Approach to Prioritize Nip...</td>\n",
       "      <td>Nipah Virus (NiV) came into limelight recentl...</td>\n",
       "      <td>Nipah is an infectious negative-sense single-...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>013d9d1cba8a54d5d3718c229b812d7cf91b6c89</td>\n",
       "      <td>Assessing spread risk of Wuhan novel coronavir...</td>\n",
       "      <td>Background: A novel coronavirus (2019-nCoV) e...</td>\n",
       "      <td>In December 2019, a cluster of patients with ...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>01d162d7fae6aaba8e6e60e563ef4c2fca7b0e18</td>\n",
       "      <td>TWIRLS, an automated topic-wise inference meth...</td>\n",
       "      <td>Faced with the current large-scale public hea...</td>\n",
       "      <td>The sudden outbreak of the new coronavirus (S...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>01e3b313e78a352593be2ff64927192af66619b5</td>\n",
       "      <td>Title: Viruses are a dominant driver of protei...</td>\n",
       "      <td>Viruses interact with hundreds to thousands o...</td>\n",
       "      <td>The copyright holder for this preprint (which...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>02201e4601ab0eb70b6c26480cf2bfeae2625193</td>\n",
       "      <td>The impact of regular school closure on season...</td>\n",
       "      <td>School closure is often considered as an opti...</td>\n",
       "      <td>they affect seasonal influenza dynamics. The ...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>0255ea4b2f26a51a3bfa3bd8f3e1978c82c976d5</td>\n",
       "      <td>Carbon Nanocarriers Deliver siRNA to Intact Pl...</td>\n",
       "      <td>Post-transcriptional gene silencing (PTGS) is...</td>\n",
       "      <td>siRNA without relying on plant transformation...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>029c1c588047f1d612a219ee15494d2d19ff7439</td>\n",
       "      <td>Protective Population Behavior Change in Outbr...</td>\n",
       "      <td></td>\n",
       "      <td>estimation is complicated, however, by effort...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>03ce432f27c7df6af22b92245a614db2ecb5de5f</td>\n",
       "      <td>A hidden gene in astroviruses encodes a cell-p...</td>\n",
       "      <td>Human astroviruses are small nonenveloped vir...</td>\n",
       "      <td>Humans astroviruses (HAstVs) belong to genus ...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>03ea3a614b56409d3f099c9ad764864293132540</td>\n",
       "      <td>Live-cell single RNA imaging reveals bursts of...</td>\n",
       "      <td>Ribosomal frameshifting during the translatio...</td>\n",
       "      <td>Frameshifting is a fundamental biological pro...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>04030bba3035a58c7725ae267973206f6eb6c0b4</td>\n",
       "      <td>Development and Evaluation of A CRISPR-based D...</td>\n",
       "      <td>The recent outbreak of infections by the 2019...</td>\n",
       "      <td>Since the beginning of 2020, a surging number...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>041bae0a6de2b69979d39460b3f2ee8946534ec2</td>\n",
       "      <td>Estimation of the final size of the second pha...</td>\n",
       "      <td>In the note, the logistic growth regression m...</td>\n",
       "      <td>however, use approximate solution and thus ob...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>05082393ba4c7ec530190dd887d99c74fd72f6d6</td>\n",
       "      <td>Self-assembly of the RZZ complex into filament...</td>\n",
       "      <td></td>\n",
       "      <td>To examine the role of the kinetochore dynein...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>05d47dd5b46f86428de058db4ecc8bca76a9ad16</td>\n",
       "      <td>Development of CRISPR as a prophylactic strate...</td>\n",
       "      <td></td>\n",
       "      <td>The world is currently faced with a pandemic ...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>05d99c07db59b6948e39bfa62c2cbbf62944059a</td>\n",
       "      <td>A spatial model of CoVID-19 transmission in En...</td>\n",
       "      <td>An outbreak of a novel coronavirus, named CoV...</td>\n",
       "      <td>An outbreak of a novel coronavirus, recently ...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>0624a12abfe85c8b5070850d912a2db4cd453236</td>\n",
       "      <td>TALC: Transcription Aware Long Read Correction</td>\n",
       "      <td>Long-read sequencing technologies are invalua...</td>\n",
       "      <td>Recent advances in RNA sequencing (RNA-seq) t...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>06a1002f9fbea7179ac3572843f66b14568af6e4</td>\n",
       "      <td>Nelfinavir was predicted to be a potential inh...</td>\n",
       "      <td>2019-nCov has caused more than 80 deaths as o...</td>\n",
       "      <td>In December 2019, cluster of patients with pn...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>06c1b3535b83251cf92c01258b5048beeab7a460</td>\n",
       "      <td>Beyond R 0 : the importance of contact tracing...</td>\n",
       "      <td>The basic reproductive number -R 0 -is one of...</td>\n",
       "      <td>The basic reproductive number -R 0 -is one of...</td>\n",
       "      <td>biorxiv_medrxiv</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    paper_id  \\\n",
       "0   0015023cc06b5362d332b3baf348d11567ca2fbb   \n",
       "1   004f0f8bb66cf446678dc13cf2701feec4f36d76   \n",
       "2   00d16927588fb04d4be0e6b269fc02f0d3c2aa7b   \n",
       "3   0139ea4ca580af99b602c6435368e7fdbefacb03   \n",
       "4   013d9d1cba8a54d5d3718c229b812d7cf91b6c89   \n",
       "5   01d162d7fae6aaba8e6e60e563ef4c2fca7b0e18   \n",
       "6   01e3b313e78a352593be2ff64927192af66619b5   \n",
       "7   02201e4601ab0eb70b6c26480cf2bfeae2625193   \n",
       "8   0255ea4b2f26a51a3bfa3bd8f3e1978c82c976d5   \n",
       "9   029c1c588047f1d612a219ee15494d2d19ff7439   \n",
       "10  03ce432f27c7df6af22b92245a614db2ecb5de5f   \n",
       "11  03ea3a614b56409d3f099c9ad764864293132540   \n",
       "12  04030bba3035a58c7725ae267973206f6eb6c0b4   \n",
       "13  041bae0a6de2b69979d39460b3f2ee8946534ec2   \n",
       "14  05082393ba4c7ec530190dd887d99c74fd72f6d6   \n",
       "15  05d47dd5b46f86428de058db4ecc8bca76a9ad16   \n",
       "16  05d99c07db59b6948e39bfa62c2cbbf62944059a   \n",
       "17  0624a12abfe85c8b5070850d912a2db4cd453236   \n",
       "18  06a1002f9fbea7179ac3572843f66b14568af6e4   \n",
       "19  06c1b3535b83251cf92c01258b5048beeab7a460   \n",
       "\n",
       "                                                title  \\\n",
       "0   The RNA pseudoknots in foot-and-mouth disease ...   \n",
       "1   Healthcare-resource-adjusted vulnerabilities t...   \n",
       "2   Real-time, MinION-based, amplicon sequencing f...   \n",
       "3   A Combined Evidence Approach to Prioritize Nip...   \n",
       "4   Assessing spread risk of Wuhan novel coronavir...   \n",
       "5   TWIRLS, an automated topic-wise inference meth...   \n",
       "6   Title: Viruses are a dominant driver of protei...   \n",
       "7   The impact of regular school closure on season...   \n",
       "8   Carbon Nanocarriers Deliver siRNA to Intact Pl...   \n",
       "9   Protective Population Behavior Change in Outbr...   \n",
       "10  A hidden gene in astroviruses encodes a cell-p...   \n",
       "11  Live-cell single RNA imaging reveals bursts of...   \n",
       "12  Development and Evaluation of A CRISPR-based D...   \n",
       "13  Estimation of the final size of the second pha...   \n",
       "14  Self-assembly of the RZZ complex into filament...   \n",
       "15  Development of CRISPR as a prophylactic strate...   \n",
       "16  A spatial model of CoVID-19 transmission in En...   \n",
       "17     TALC: Transcription Aware Long Read Correction   \n",
       "18  Nelfinavir was predicted to be a potential inh...   \n",
       "19  Beyond R 0 : the importance of contact tracing...   \n",
       "\n",
       "                                             abstract  \\\n",
       "0    word count: 194 22 Text word count: 5168 23 2...   \n",
       "1                                                       \n",
       "2    Infectious bronchitis (IB) causes significant...   \n",
       "3    Nipah Virus (NiV) came into limelight recentl...   \n",
       "4    Background: A novel coronavirus (2019-nCoV) e...   \n",
       "5    Faced with the current large-scale public hea...   \n",
       "6    Viruses interact with hundreds to thousands o...   \n",
       "7    School closure is often considered as an opti...   \n",
       "8    Post-transcriptional gene silencing (PTGS) is...   \n",
       "9                                                       \n",
       "10   Human astroviruses are small nonenveloped vir...   \n",
       "11   Ribosomal frameshifting during the translatio...   \n",
       "12   The recent outbreak of infections by the 2019...   \n",
       "13   In the note, the logistic growth regression m...   \n",
       "14                                                      \n",
       "15                                                      \n",
       "16   An outbreak of a novel coronavirus, named CoV...   \n",
       "17   Long-read sequencing technologies are invalua...   \n",
       "18   2019-nCov has caused more than 80 deaths as o...   \n",
       "19   The basic reproductive number -R 0 -is one of...   \n",
       "\n",
       "                                                 body           origin  \n",
       "0    VP3, and VP0 (which is further processed to V...  biorxiv_medrxiv  \n",
       "1    The 2019-nCoV epidemic has spread across Chin...  biorxiv_medrxiv  \n",
       "2    Infectious bronchitis (IB), which is caused b...  biorxiv_medrxiv  \n",
       "3    Nipah is an infectious negative-sense single-...  biorxiv_medrxiv  \n",
       "4    In December 2019, a cluster of patients with ...  biorxiv_medrxiv  \n",
       "5    The sudden outbreak of the new coronavirus (S...  biorxiv_medrxiv  \n",
       "6    The copyright holder for this preprint (which...  biorxiv_medrxiv  \n",
       "7    they affect seasonal influenza dynamics. The ...  biorxiv_medrxiv  \n",
       "8    siRNA without relying on plant transformation...  biorxiv_medrxiv  \n",
       "9    estimation is complicated, however, by effort...  biorxiv_medrxiv  \n",
       "10   Humans astroviruses (HAstVs) belong to genus ...  biorxiv_medrxiv  \n",
       "11   Frameshifting is a fundamental biological pro...  biorxiv_medrxiv  \n",
       "12   Since the beginning of 2020, a surging number...  biorxiv_medrxiv  \n",
       "13   however, use approximate solution and thus ob...  biorxiv_medrxiv  \n",
       "14   To examine the role of the kinetochore dynein...  biorxiv_medrxiv  \n",
       "15   The world is currently faced with a pandemic ...  biorxiv_medrxiv  \n",
       "16   An outbreak of a novel coronavirus, recently ...  biorxiv_medrxiv  \n",
       "17   Recent advances in RNA sequencing (RNA-seq) t...  biorxiv_medrxiv  \n",
       "18   In December 2019, cluster of patients with pn...  biorxiv_medrxiv  \n",
       "19   The basic reproductive number -R 0 -is one of...  biorxiv_medrxiv  "
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## We have 29,315 rows and 5 columns\n",
    "print(df.shape)\n",
    "df.head(20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Exporting this to a csv\n",
    "\n",
    "df.to_csv(\"COVID-19 journals with NaNs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Investigating missing values\n",
    "df_missing_body = df['body'].replace(\"\", np.nan)\n",
    "df_missing_body = df_missing_body.dropna()\n",
    "\n",
    "df_missing_abstract = df['abstract'].replace(\"\", np.nan)\n",
    "df_missing_abstract = df_missing_abstract.dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(29315,)\n",
      "(21264,)\n"
     ]
    }
   ],
   "source": [
    "## All of our files have a journal/body text, but not of them have abstracts. 8,051 are missing abstracts\n",
    "print(df_missing_body.shape)\n",
    "print(df_missing_abstract.shape)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\n"
     ]
    }
   ],
   "source": [
    "%cd \"C:\\Users\\Jordan\\Desktop\\COVID-19 Challenge\"\n",
    "\n",
    "metadata_df = pd.read_csv(\"metadata.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(44220, 15)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>journal</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "      <th>full_text_file</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Intrauterine virus infections and congenital h...</td>\n",
       "      <td>10.1016/0002-8703(72)90077-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4361535.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract The etiologic basis for the vast majo...</td>\n",
       "      <td>1972-12-31</td>\n",
       "      <td>Overall, James C.</td>\n",
       "      <td>American Heart Journal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Coronaviruses in Balkan nephritis</td>\n",
       "      <td>10.1016/0002-8703(80)90355-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6243850.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1980-03-31</td>\n",
       "      <td>Georgescu, Leonida; Diosi, Peter; Buţiu, Ioan;...</td>\n",
       "      <td>American Heart Journal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Cigarette smoking and coronary heart disease: ...</td>\n",
       "      <td>10.1016/0002-8703(80)90356-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7355701.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1980-03-31</td>\n",
       "      <td>Friedman, Gary D</td>\n",
       "      <td>American Heart Journal</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>aecbc613ebdab36753235197ffb4f35734b5ca63</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Clinical and immunologic studies in identical ...</td>\n",
       "      <td>10.1016/0002-9343(73)90176-9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4579077.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract Middle-aged female identical twins, o...</td>\n",
       "      <td>1973-08-31</td>\n",
       "      <td>Brunner, Carolyn M.; Horwitz, David A.; Shann,...</td>\n",
       "      <td>The American Journal of Medicine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Epidemiology of community-acquired respiratory...</td>\n",
       "      <td>10.1016/0002-9343(85)90361-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4014285.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract Upper respiratory tract infections ar...</td>\n",
       "      <td>1985-06-28</td>\n",
       "      <td>Garibaldi, Richard A.</td>\n",
       "      <td>The American Journal of Medicine</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        sha  source_x  \\\n",
       "0                                       NaN  Elsevier   \n",
       "1                                       NaN  Elsevier   \n",
       "2                                       NaN  Elsevier   \n",
       "3  aecbc613ebdab36753235197ffb4f35734b5ca63  Elsevier   \n",
       "4                                       NaN  Elsevier   \n",
       "\n",
       "                                               title  \\\n",
       "0  Intrauterine virus infections and congenital h...   \n",
       "1                  Coronaviruses in Balkan nephritis   \n",
       "2  Cigarette smoking and coronary heart disease: ...   \n",
       "3  Clinical and immunologic studies in identical ...   \n",
       "4  Epidemiology of community-acquired respiratory...   \n",
       "\n",
       "                            doi pmcid  pubmed_id    license  \\\n",
       "0  10.1016/0002-8703(72)90077-4   NaN  4361535.0  els-covid   \n",
       "1  10.1016/0002-8703(80)90355-5   NaN  6243850.0  els-covid   \n",
       "2  10.1016/0002-8703(80)90356-7   NaN  7355701.0  els-covid   \n",
       "3  10.1016/0002-9343(73)90176-9   NaN  4579077.0  els-covid   \n",
       "4  10.1016/0002-9343(85)90361-4   NaN  4014285.0  els-covid   \n",
       "\n",
       "                                            abstract publish_time  \\\n",
       "0  Abstract The etiologic basis for the vast majo...   1972-12-31   \n",
       "1                                                NaN   1980-03-31   \n",
       "2                                                NaN   1980-03-31   \n",
       "3  Abstract Middle-aged female identical twins, o...   1973-08-31   \n",
       "4  Abstract Upper respiratory tract infections ar...   1985-06-28   \n",
       "\n",
       "                                             authors  \\\n",
       "0                                  Overall, James C.   \n",
       "1  Georgescu, Leonida; Diosi, Peter; Buţiu, Ioan;...   \n",
       "2                                   Friedman, Gary D   \n",
       "3  Brunner, Carolyn M.; Horwitz, David A.; Shann,...   \n",
       "4                              Garibaldi, Richard A.   \n",
       "\n",
       "                            journal  Microsoft Academic Paper ID  \\\n",
       "0            American Heart Journal                          NaN   \n",
       "1            American Heart Journal                          NaN   \n",
       "2            American Heart Journal                          NaN   \n",
       "3  The American Journal of Medicine                          NaN   \n",
       "4  The American Journal of Medicine                          NaN   \n",
       "\n",
       "  WHO #Covidence  has_full_text  full_text_file  \n",
       "0            NaN          False  custom_license  \n",
       "1            NaN          False  custom_license  \n",
       "2            NaN          False  custom_license  \n",
       "3            NaN           True  custom_license  \n",
       "4            NaN          False  custom_license  "
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(metadata_df.shape)\n",
    "metadata_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(28462, 15)"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "##The True/False column states that There are 28,462 records with \n",
    "metadata_df[metadata_df['has_full_text'] == True].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Get list of ids "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Get list of ids \n",
    "sha_list = metadata_df['sha'].dropna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "28462"
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## This is the same length as those listed as True...\n",
    "len(sha_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27690 from our ids are in the metadata ids\n",
      "27678 from the metadata ids are in our ids\n"
     ]
    }
   ],
   "source": [
    "## Testing to see whats's here\n",
    "\n",
    "\n",
    "## 27,690 values from  our list of ids is in the metadata\n",
    "print(str(sha_list.isin(paper_id).sum()) + \" from our ids are in the metadata ids\")\n",
    "\n",
    "## Meanwhile, 27,678 values from the metadata ids are in our paper ids\n",
    "print(str(df['paper_id'].isin(sha_list).sum()) + \" from the metadata ids are in our ids\")\n",
    "\n",
    "## There appears to be a difference of 22 ids that we have that aren't in the metdata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [],
   "source": [
    "##Creating dataframe of the two dataframes merged. Doing an outer merge so that we can see the overlap\n",
    "merged_df = pd.merge(metadata_df, df, how = 'outer', left_on = \"sha\", right_on ='paper_id', indicator = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(45857, 21)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sha</th>\n",
       "      <th>source_x</th>\n",
       "      <th>title_x</th>\n",
       "      <th>doi</th>\n",
       "      <th>pmcid</th>\n",
       "      <th>pubmed_id</th>\n",
       "      <th>license</th>\n",
       "      <th>abstract_x</th>\n",
       "      <th>publish_time</th>\n",
       "      <th>authors</th>\n",
       "      <th>...</th>\n",
       "      <th>Microsoft Academic Paper ID</th>\n",
       "      <th>WHO #Covidence</th>\n",
       "      <th>has_full_text</th>\n",
       "      <th>full_text_file</th>\n",
       "      <th>paper_id</th>\n",
       "      <th>title_y</th>\n",
       "      <th>abstract_y</th>\n",
       "      <th>body</th>\n",
       "      <th>origin</th>\n",
       "      <th>_merge</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Intrauterine virus infections and congenital h...</td>\n",
       "      <td>10.1016/0002-8703(72)90077-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4361535.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract The etiologic basis for the vast majo...</td>\n",
       "      <td>1972-12-31</td>\n",
       "      <td>Overall, James C.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Coronaviruses in Balkan nephritis</td>\n",
       "      <td>10.1016/0002-8703(80)90355-5</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6243850.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1980-03-31</td>\n",
       "      <td>Georgescu, Leonida; Diosi, Peter; Buţiu, Ioan;...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Cigarette smoking and coronary heart disease: ...</td>\n",
       "      <td>10.1016/0002-8703(80)90356-7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7355701.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1980-03-31</td>\n",
       "      <td>Friedman, Gary D</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Epidemiology of community-acquired respiratory...</td>\n",
       "      <td>10.1016/0002-9343(85)90361-4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4014285.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract Upper respiratory tract infections ar...</td>\n",
       "      <td>1985-06-28</td>\n",
       "      <td>Garibaldi, Richard A.</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NaN</td>\n",
       "      <td>Elsevier</td>\n",
       "      <td>Monoclonal antibodies identify multiple epitop...</td>\n",
       "      <td>10.1016/0006-291X(85)91946-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2409966.0</td>\n",
       "      <td>els-covid</td>\n",
       "      <td>Abstract Nine hybridoma cell lines secreting a...</td>\n",
       "      <td>1985-06-28</td>\n",
       "      <td>Cherel, Isabelle; Grosclaude, Jeanne; Rouze, P...</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>custom_license</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>left_only</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   sha  source_x                                            title_x  \\\n",
       "0  NaN  Elsevier  Intrauterine virus infections and congenital h...   \n",
       "1  NaN  Elsevier                  Coronaviruses in Balkan nephritis   \n",
       "2  NaN  Elsevier  Cigarette smoking and coronary heart disease: ...   \n",
       "3  NaN  Elsevier  Epidemiology of community-acquired respiratory...   \n",
       "4  NaN  Elsevier  Monoclonal antibodies identify multiple epitop...   \n",
       "\n",
       "                            doi pmcid  pubmed_id    license  \\\n",
       "0  10.1016/0002-8703(72)90077-4   NaN  4361535.0  els-covid   \n",
       "1  10.1016/0002-8703(80)90355-5   NaN  6243850.0  els-covid   \n",
       "2  10.1016/0002-8703(80)90356-7   NaN  7355701.0  els-covid   \n",
       "3  10.1016/0002-9343(85)90361-4   NaN  4014285.0  els-covid   \n",
       "4  10.1016/0006-291X(85)91946-1   NaN  2409966.0  els-covid   \n",
       "\n",
       "                                          abstract_x publish_time  \\\n",
       "0  Abstract The etiologic basis for the vast majo...   1972-12-31   \n",
       "1                                                NaN   1980-03-31   \n",
       "2                                                NaN   1980-03-31   \n",
       "3  Abstract Upper respiratory tract infections ar...   1985-06-28   \n",
       "4  Abstract Nine hybridoma cell lines secreting a...   1985-06-28   \n",
       "\n",
       "                                             authors  ...  \\\n",
       "0                                  Overall, James C.  ...   \n",
       "1  Georgescu, Leonida; Diosi, Peter; Buţiu, Ioan;...  ...   \n",
       "2                                   Friedman, Gary D  ...   \n",
       "3                              Garibaldi, Richard A.  ...   \n",
       "4  Cherel, Isabelle; Grosclaude, Jeanne; Rouze, P...  ...   \n",
       "\n",
       "  Microsoft Academic Paper ID  WHO #Covidence has_full_text  full_text_file  \\\n",
       "0                         NaN             NaN         False  custom_license   \n",
       "1                         NaN             NaN         False  custom_license   \n",
       "2                         NaN             NaN         False  custom_license   \n",
       "3                         NaN             NaN         False  custom_license   \n",
       "4                         NaN             NaN         False  custom_license   \n",
       "\n",
       "  paper_id title_y abstract_y body origin     _merge  \n",
       "0      NaN     NaN        NaN  NaN    NaN  left_only  \n",
       "1      NaN     NaN        NaN  NaN    NaN  left_only  \n",
       "2      NaN     NaN        NaN  NaN    NaN  left_only  \n",
       "3      NaN     NaN        NaN  NaN    NaN  left_only  \n",
       "4      NaN     NaN        NaN  NaN    NaN  left_only  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## We went from (44220, 15) to (45857, 21)\n",
    "print(merged_df.shape)\n",
    "\n",
    "merged_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "both          27690\n",
       "left_only     16530\n",
       "right_only     1637\n",
       "Name: _merge, dtype: int64"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## Creating a value counts to see spread. 27690 exist in both, 16530 \"exist\" in the metadata only, and 1637 only exist in\n",
    "## our scraped paper ids from the files. Interesting. \n",
    "merged_df['_merge'].value_counts()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
